### PR TITLE
Completes Task 4

### PR DESCRIPTION
**Pull request by tylerkade**
No styling as of yet.

### CODE REVIEW

File: api.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import axios from "axios";
+
+const api = axios.create({
+  baseURL: "https://my-nc-news-t13l.onrender.com/api",
+});
+
+const getArticles = () => {
+  let url = "/articles";
+  return api.get(url).then(({ data }) => {
+    return data.articles;
```
Comment (nc-mohammed-dastgir) [inline api.js:10]: Nice work extracting out this logic into a separate file, but url should not be defined with a let as this is not something we want to really change.

File: src/components/ArticleList.jsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  
+
+  return (
+    <>
+      <ul className="article-list">
+        {articles.map((article) => {
+          return <ArticleCard key={article.article_id} article={article}/>;
+        })}
+      </ul>
+    </>
```
Comment (nc-mohammed-dastgir) [inline src/components/ArticleList.jsx:41]: This works, but it doesnt seem to be following your plan as the plan suggests there should be an ArticlesList component here which renders the ArticleCard components

File: src/components/ArticleList.jsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        }
+      })
+      .catch(() => {
+        setIsLoading(false);
+      });
+  }, []);
+
+  if (isLoading) {
+    return <div role="alert">Loading Articles...</div>;
+  }
```
Comment (nc-mohammed-dastgir) [inline src/components/ArticleList.jsx:29]: I am a big fan of the loading logic, but should the default loading value be false? Are we assuming on initial load we actually have the data already?

File: src/components/ArticleList.jsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+  // console.log(articles);
```
Comment (nc-mohammed-dastgir) [inline src/components/ArticleList.jsx:31]: Dont forget to remove console.log's and un-needed comments before pushing your work up to github

File: src/components/ArticleCard.jsx [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    
+  return (
+    <div className="article-card">
+      <li>
+        <a href={`/articles/${article.topic}/${article.article_id}`}>
+          <h3>{article.title}</h3>
+        </a>
+      </li>
+    </div>
+  );
```
Comment (nc-mohammed-dastgir) [inline src/components/ArticleCard.jsx:12]: This does not follow the plan either as it is just loading a single title and nothing more. You dont need to finish every ticket looking perfect, but I would recommend using all available content and displaying this on the page at the very least.

**Review by nc-mohammed-dastgir — COMMENTED** [review]
This is a good start, but do have a read of my below comments as well:

File: src/components/ArticleCard.jsx (hunk 5 above)
Comment (tylerkade) [reply src/components/ArticleCard.jsx:12]: The original plan has images, but this data (the array with all articles) doesn't contain the article images, hence why it looks different. The only data in this is article_id, author, comment_count, created_at, title, topic, and votes.

**Pull request merged**